lightning/src/ln/peer_handler.rs

```diff
@@ -2396,6 +2400,13 @@ impl<Descriptor: SocketDescriptor, CM: Deref, RM: Deref, OM: Deref, L: Deref, CM
 				}
 			}
 		}
+
+		let nodes_to_connect =
+			self.message_handler.onion_message_handler.release_pending_connections();
+		let mut pending_events = self.pending_events.lock().unwrap();
+		for node_announcement in nodes_to_connect {
+			pending_events.push(Event::ConnectionNeeded { node_announcement });
+		}
```

Comment:
Originally, I had this in `process_events`, but I moved it to `timer_tick_occurred` so the timeout could be in terms of ticks. However, it has occurred to me that having it here means we may wait up to 10 seconds before even generating the events, which would result in bad UX. We could instead simply generate events directly in `process_pending_events` and `process_pending_events_async`, dropping the intermediary `pending_events` `Vec`. But, as before, we'd have to define the timeout in something other than ticks, or possibly add another method to `OnionMessageHandler` for calling in `timer_tick_occurred`.

Reply:
Updated the `OnionMessageHandler` trait to have a `timer_tick_occurred` method so that dropping buffered messages is independent of providing events.